### ISSUE TITLE
Перевернуто изображение

### ISSUE THREAD

**Issue by Nikola131313** [Kazakov1988/All-terrain-vehicle#1]
Изображение с камеры повернуто на 90 градусов.

**Comment by Kazakov1988** [comment]
> 

Переверни модуль )

**Comment by AdUser** [comment]
Тоже столкнулся с повёрнутым изображением. Народ [пишет](https://reddit.com/r/esp32/comments/e411yo/esp32cam_rotating_picture_90/) что есть минимум две версии модуля OV2640, у одного из которых изображение "вертикальное", а у другого - "горизонтальное". Программно это исправить сложно - камера умеет только vflip и mirror, со стороны html повернуть можно через css `transform: rotate(270deg)`, но "едет" верстка и начинаются [сложности](https://stackoverflow.com/questions/43143605/css-background-image-rotate-without-transform-rotate) с рассчётом bound-box'ов.
Мне кажется самый простой способ - перемоделировать одну детальку крепления модуля.

**Comment by Kazakov1988** [comment]
> Тоже столкнулся с повёрнутым изображением. Народ [пишет](https://reddit.com/r/esp32/comments/e411yo/esp32cam_rotating_picture_90/) что есть минимум две версии модуля OV2640, у одного из которых изображение "вертикальное", а у другого - "горизонтальное". Программно это исправить сложно - камера умеет только vflip и mirror, со стороны html повернуть можно через css `transform: rotate(270deg)`, но "едет" верстка и начинаются [сложности](https://stackoverflow.com/questions/43143605/css-background-image-rotate-without-transform-rotate) с рассчётом bound-box'ов. Мне кажется самый простой способ - перемоделировать одну детальку крепления модуля.

Да, так и есть. Можно ещё 2-3 модуля таких купить и один хотя бы подойти должен. Они не дорогие.

**Comment by AdUser** [comment]
Я [сделал](http://linuxdv.org/filebin/1683873833) две альтернативных модели крепления `ESP32_CAM_holder_A`, развернутых на 90 и 270 градусов соответственно. Крышку переделывать нет необходимости, она подходит и к новым моделям. В архиве две пары .stl и .step

Скачайте себе куда-нибудь и дополните проект на thingiverse (через месяц мой файл удалится, а гитхаб не даёт грузить архивы и бинарники).

**Comment by Kazakov1988** [comment]
> Я [сделал](http://linuxdv.org/filebin/1683873833) две альтернативных модели крепления `ESP32_CAM_holder_A`, развернутых на 90 и 270 градусов соответственно. Крышку переделывать нет необходимости, она подходит и к новым моделям. В архиве две пары .stl и .step
> 
> Скачайте себе куда-нибудь и дополните проект на thingiverse (через месяц мой файл удалится, а гитхаб не даёт грузить архивы и бинарники).

Готово - https://www.thingiverse.com/thing:4915508 Кому-то пригодится.
